src/backend/api/reviews.js

```diff
@@ -0,0 +1,117 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+const reviewsColumns = new Set([
+  "title",
+  "description",
+  "meal_id",
+  "stars",
+  "created_date",
+]);
+
+// this function checks if input data valid and makes possible to update only neccesary data in table
+function checkTableData(table) {
+  try {
+    let validTableData = {};
+    for (const key in table) {
+      if (reviewsColumns.has(key)) {
+        validTableData[key] = table[key];
+      }
+    }
+    if ("meal_id" in validTableData) {
+      if (isNaN(Number(validTableData.meal_id))) {
+        throw new Error();
+      }
+    }
+
+    if ("stars" in validTableData) {
+      if (isNaN(Number(validTableData.stars))) {
+        throw new Error();
+      }
+    }
+    if ("created_date" in validTableData) {
+      const createdDateFormatted = new Date(validTableData.created_date);
+      if (!Date.parse(createdDateFormatted)) {
+        throw new Error();
```

Comment:
How will you tell the difference between the errors that come from line 24, 30 and 36?